File: spark-using-java/src/main/java/spark/examples/SparkContextExample.java
package spark.examples;

import org.apache.spark.api.java.JavaSparkContext;

public class SparkContextExample {

	public static void main(String[] args) {
		String master = System.getenv("MASTER");
		if (master == null) {
			master = "local";
		}
		String sparkHome = System.getenv("SPARK_HOME");
		if (sparkHome == null) {
			sparkHome = "./";
		}
		String jars = System.getenv("JARS");

		JavaSparkContext ctx = new JavaSparkContext(master, "my Java app", sparkHome , jars);
		
		System.out.println(ctx.getConf().toDebugString());
		
		ctx.stop();
	}

}

File: spark-using-scala/src/main/scala/scala/examples/SparkContextExample.scala
package scala.examples

import org.apache.spark.SparkContext
import scala.util.Properties

object SparkContextExample {
  def main(args: Array[String]): Unit = {
    val master = Properties.envOrElse("MASTER", "spark://PCW1844946L:7077")
    val sparkHome = Properties.propOrElse("SPARK_HOME", "")
    val myJars = Seq(java.lang.System.getenv(("JARS")))
    val sparkContext = new SparkContext(master, "my app", sparkHome, myJars)
    
    println(sparkContext.getConf.toDebugString)
    
    sparkContext.stop
  }
}